### ISSUE TITLE
Voicing delayed or absent with Chromebook

### ISSUE THREAD

**Issue by Nancy-Salpepi** [phetsims/gravity-force-lab-basics#303]
**Test device**
Chromebook

**Browser**
Chrome

**Problem description**
https://github.com/phetsims/qa/issues/702

Voicing is delayed or requires more than 1 click sometimes to work. Once it starts working, the delay is lessened. 

**Steps to reproduce**
Here is one example: Using a chromebook--turn on voicing in preferences. You should get a delay here.

**Visuals**

https://user-images.githubusercontent.com/87318828/131896071-d9d3b70d-b5bd-4ea5-9c51-e6891fd0c627.mov

https://user-images.githubusercontent.com/87318828/131896101-4ef5e2a0-2e71-460a-981c-9a21fcb2298a.mov

Troubleshooting information:
!!!!! DO NOT EDIT !!!!!
Name: ‪Gravity Force Lab: Basics‬
URL: https://phet-dev.colorado.edu/html/gravity-force-lab-basics/1.1.0-rc.2/phet/gravity-force-lab-basics_all_phet.html
Version: 1.1.0-rc.2 2021-08-30 16:27:03 UTC
Features missing: applicationcache, applicationcache, touch
User Agent: Mozilla/5.0 (X11; CrOS x86_64 13982.82.0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.157 Safari/537.36
Language: en-US
Window: 1357x609
Pixel Ratio: 1/1
WebGL: WebGL 1.0 (OpenGL ES 2.0 Chromium)
GLSL: WebGL GLSL ES 1.0 (OpenGL ES GLSL ES 1.0 Chromium)
Vendor: WebKit (WebKit WebGL)
Vertex: attribs: 16 varying: 32 uniform: 4096
Texture: size: 16384 imageUnits: 32 (vertex: 32, combined: 192)
Max viewport: 16384x16384
OES_texture_float: true

**Comment by jessegreenberg** [comment]
Thanks @Nancy-Salpepi - would you be able to see if this is happening in the published version of john-travoltage too? I don't have access to a chromebook so I need to think about how I can investigate further.

**Comment by Nancy-Salpepi** [comment]
@jessegreenberg I saw 2 things with the published John Travoltage sim:

1. If I move between the quick info options while one is being read aloud, voicing does not tell me what option I am now on. If I interrupt the voicing after moving the arm or leg and switch focus between them, voicing works as it should. 

2. When turning on voicing for the first time, it takes 2 seconds for voicing to actually start. However, on subsequent attempts, it happens without a delay. 

https://user-images.githubusercontent.com/87318828/132554329-effd3ee9-ade7-4408-a69d-60685ecfef8d.mov


https://user-images.githubusercontent.com/87318828/132554381-94e4fd49-34ed-4020-8980-6a8effaf8d34.mov

**Comment by jessegreenberg** [comment]
@Nancy-Salpepi helped me remote access into a chromebook. I cannot hear output with remote access but I can see when the voicing responses are coming out with `?printVoicingResponses`. `?printVoicingResponses` puts the output text in the console when the actual utterance from the browser starts speaking, so it is a good indication of the output. Using this I can confirm that it is very delayed the first time voicing is used, it usually takes 5-6 seconds to produce speech.

**Comment by jessegreenberg** [comment]
Just noting some thoughts about things we could improve (or not):
1) Can we reuse the `speechSynthesisUtterance` so that we don't recreate one every speech? Maybe this will be faster, maybe not.
2) What if on the first input to the sim we trigger some empty speech so our first speech synthesis isn't the one used to say that voicing is enabled. This would kind of be like "revving the engine". 
3) I checked whether we were sending too many utterances to the queue causing a delay. In these cases in this issue we are not.
4) I thought about using different voices but wasn't really able to test this against the issue presented here because in order to change the voice I need to enable voicing first. I also can't hear if the different voices render faster.
5) I verified that the first Voicing output happens synchronously with input. Having the first output done asynchronously has been problematic for us in the past so I wanted to check this wasn't still an issue.

Option 2 seems the most promising, ill give that a go.

**Comment by jessegreenberg** [comment]
Here is the patch to try:

```patch
Index: js/accessibility/voicing/voicingManager.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- js/accessibility/voicing/voicingManager.js	(revision d73d6fc643306e149c3eede8706e6d107c978397)
+++ js/accessibility/voicing/voicingManager.js	(date 1632352884035)
@@ -21,6 +21,7 @@
 import stripEmbeddingMarks from '../../../../phet-core/js/stripEmbeddingMarks.js';
 import Announcer from '../../../../utterance-queue/js/Announcer.js';
 import Utterance from '../../../../utterance-queue/js/Utterance.js';
+import Display from '../../display/Display.js';
 import scenery from '../../scenery.js';
 import globalKeyStateTracker from '../globalKeyStateTracker.js';
 import KeyboardUtils from '../KeyboardUtils.js';
@@ -173,6 +174,12 @@
       }
     } );
 
+    const engineRevListener = () => {
+      this.speakIgnoringEnabled( new Utterance( { alert: ' ' } ) );
+      Display.userGestureEmitter.removeListener( engineRevListener );
+    };
+    Display.userGestureEmitter.addListener( engineRevListener );
+
     this.initialized = true;
   }
 
@@ -337,6 +344,8 @@
 
     if ( !this.hasSpoken ) {
 
+      console.log( 'speaking' );
+
       // for the first time speaking it must be synchronous and we cannot use TimeoutCallbackObject workarounds yet
       this.getSynth().speak( speechSynthUtterance );
       this.hasSpoken = true;
Index: js/input/Input.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- js/input/Input.js	(revision d73d6fc643306e149c3eede8706e6d107c978397)
+++ js/input/Input.js	(date 1632352810206)
@@ -1051,7 +1051,9 @@
    * @param {boolean} bubbles
    */
   dispatchPDOMEvent( trail, eventType, domEvent, bubbles ) {
-    Display.userGestureEmitter.emit();
+    if ( eventType !== 'focus' && eventType !== 'focusin' ) {
+      Display.userGestureEmitter.emit();
+    }
 
     // This workaround hopefully won't be here forever, see ParallelDOM.setExcludeLabelSiblingFromInput() and https://github.com/phetsims/a11y-research/issues/156
     if ( !( domEvent.target && domEvent.target.hasAttribute( PDOMUtils.DATA_EXCLUDE_FROM_INPUT ) ) ) {

```

The `focus`  event (from scripting focus on the screen div) is triggering `Display.userGestureEmitter`, so I had to prevent that from happening to get the input gesture from actual input.

**Comment by jessegreenberg** [comment]
This is really good information, next is to decide the best way to apply it.
